<a href="https://colab.research.google.com/github/binteaamer/FlyrankInternship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/binteaamer/FlyrankInternship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [10]:
!git clone https://github.com/binteaamer/FlyrankInternship.git
%cd FlyrankInternship

Cloning into 'FlyrankInternship'...
remote: Enumerating objects: 132, done.
remote: Counting objects: 100% (132/132), done.
remote: Compressing objects: 100% (88/88), done.
remote: Total 132 (delta 44), reused 95 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (132/132), 1.85 MiB | 7.27 MiB/s, done.
Resolving deltas: 100% (44/44), done.
/content/FlyrankInternship


## 1. My lane (or freestyle) and why

Freestyle

Research question: Can average search position help identify pages with low click-through rates (CTR) that should be prioritised for SEO optimisation?

I chose this question because both average search position and CTR are available in the starter dataset. Understanding their relationship may help prioritise pages for optimisation instead of manually reviewing thousands of pages.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

Decision

Which pages should be prioritised for SEO optimisation?

Who acts?

An SEO specialist or content editor reviews the prioritised pages and decides whether to improve page titles, meta descriptions, or content.

Cost of a wrong recommendation

Prioritising the wrong pages wastes optimisation time and may leave pages with greater improvement potential untreated.

Why data helps

The dataset contains 30,000 pages with multiple SEO signals. It is difficult to manually identify patterns across thousands of pages, while data analysis can highlight relationships that support prioritisation.

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

To understand whether this question is worth investigating, I measured the relationship between average search position and click-through rate and summarised both variables using the starter dataset.

In [13]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))

corr = df["avg_position"].corr(df["ctr"])
print(f"Correlation between avg_position and CTR: {corr:.3f}")

print("\nAverage CTR:", round(df["ctr"].mean(),3))
print("Average search position:", round(df["avg_position"].mean(),2))

visible = df[df["impressions_90d"] >= 100]

print("\nAverage CTR by position tier:")
print(
    visible.groupby("position_tier")["ctr"]
    .mean()
    .round(3)
)

Total pages: 30000
Correlation between avg_position and CTR: -0.073

Average CTR: 0.511
Average search position: 16.34

Average CTR by position tier:
position_tier
deep        0.055
page_1      0.355
page_3_5    0.142
striking    0.256
top_3       0.334
Name: ctr, dtype: float64


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful words

This project measures observed relationships between average search position and click-through rate. The results are intended to support SEO prioritisation decisions.

This work cannot prove that search position causes changes in CTR or explain Google's ranking algorithm. Any conclusions will describe observed patterns in the available dataset.

The observed correlation between average search position and CTR was -0.073, indicating a weak negative relationship. Average search position alone does not strongly explain differences in CTR.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.